pkg/prometheus_ingester/prometheus_ingester.go

```diff
@@ -75,9 +76,48 @@ type queryOptions struct {
 	ResultAsQuantity *bool
 }
 
+type PrometheusIngesterHttpHeader struct {
+	Name            string
+	ValueFromString string `yaml:"valueFromString"`
```

Comment:
Why not keep it simple here with just two fields?

The YAML is unmarshalled to an empty struct -- we can detect when the user set up both. I suggest to keep "Value" and "ValueFromEnv".

Reply:
fixed d48902d4ad77ea48e830ce5ce24c2a6aa5fce662